pontoon/insights/migrations/0003_project_insights.py

```diff
@@ -0,0 +1,199 @@
+# Generated by Django 3.2.4 on 2021-08-24 06:45
+
+import datetime
+from django.db import migrations, models
+import pontoon.insights.models
+
+
+_activity_sql = """
+WITH filt AS (
+  WITH
+    sync AS (SELECT id FROM auth_user WHERE email = 'pontoon-sync@example.com')
+  SELECT
+    act.created_at::date AS created_at,
+    tra.locale_id,
+    res.project_id,
+    action_type = 'translation:created' AND cardinality(tra.machinery_sources) = 0 AS human_translation,
+    action_type = 'translation:created' AND cardinality(tra.machinery_sources) != 0 AS machinery_translation,
+    act.performed_by_id = sync.id AS is_sync,
+    act.action_type = 'translation:created' AND (tra.approved_date IS NULL OR tra.approved_date > tra.date) AS new_suggestion,
+    (
+      (action_type = 'translation:approved' AND act.performed_by_id = tra.user_id) OR
+      (action_type = 'translation:created' AND act.performed_by_id = tra.approved_user_id)
+    ) AS self_approved,
+    action_type = 'translation:approved' AND act.performed_by_id != tra.user_id AS peer_approved,
+    action_type = 'translation:rejected' AS rejected
+  FROM
+    sync,
+    actionlog_actionlog AS act,
+    base_translation AS tra,
+    base_entity AS ent,
+    base_resource AS res
+  WHERE
+    act.translation_id = tra.id AND
+    tra.entity_id = ent.id AND
+    ent.resource_id = res.id
+), sums AS (
+  SELECT
+    created_at,
+    project_id,
+    count(*) FILTER (WHERE filt.human_translation) AS human_translations,
+    count(*) FILTER (WHERE filt.machinery_translation) AS machinery_translations,
+    count(*) FILTER (WHERE NOT is_sync AND filt.new_suggestion) AS new_suggestions,
+    count(*) FILTER (WHERE NOT is_sync AND filt.self_approved) AS self_approved,
+    count(*) FILTER (WHERE NOT is_sync AND filt.peer_approved) AS peer_approved,
+    count(*) FILTER (WHERE NOT is_sync AND filt.rejected) AS rejected
+  FROM insights_projectinsightssnapshot JOIN filt USING (created_at, project_id)
+  GROUP BY created_at, project_id
+)
+UPDATE insights_projectinsightssnapshot AS ins
+SET
+  human_translations = sums.human_translations,
+  machinery_translations = sums.machinery_translations,
+  new_suggestions = sums.new_suggestions,
+  self_approved = sums.self_approved,
```

Comment:
@mathjazz Given that this is a one-off operation, does it really make sense to rewrite it using Django ORM? My suspicion while working on this was that the views of the data that are being used by this `UPDATE` might require some custom ORM models to be used to represent the data, and that it would indeed be quite a bit more work to represent the data there. The performance difference is probably measurable, but irrelevant as this only needs to be done once.

@jotes I have not experimented yet with window functions. But they do look rather interesting!

Reply:
`F()` and `Func()` sound even less readable. :) So let's stick to SQL, especially now that we already have it.